src/geometry.rs

```diff
@@ -460,25 +615,18 @@ fn n(x: f32, y: f32, z: f32) -> Vector3<f32> {
     Vector3::new(x, y, z)
 }
 
-fn tf_vn(i1: u32, i2: u32, i3: u32) -> TriangleFace {
-    TriangleFace {
-        vertices: (i1, i2, i3),
-        normals: Some((i1, i2, i3)),
-    }
-}
+fn compute_triangle_normal(p1: &Point3<f32>, p2: &Point3<f32>, p3: &Point3<f32>) -> Vector3<f32> {
+    let u = p2 - p1;
+    let v = p3 - p1;
 
-fn tf_vn_separate(v1: u32, v2: u32, v3: u32, n1: u32, n2: u32, n3: u32) -> TriangleFace {
-    TriangleFace {
-        vertices: (v1, v2, v3),
-        normals: Some((n1, n2, n3)),
-    }
+    Vector3::cross(&u, &v)
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    fn quad() -> (Vec<TriangleFace>, Vec<Point3<f32>>) {
+    fn quad() -> (Vec<(u32, u32, u32)>, Vec<Point3<f32>>) {
```

Comment:
Clippy actually complains about complexity of this type, but since it's in tests it probably doesn't matter that much.

Reply:
Yeah, it's purely anonymous test code